### DISCUSSION TITLE
转Inference模型时发生错误：TypeError: can only concatenate list (not "NoneType") to list

### DISCUSSION THREAD

**Discussion by chunmengtu** [PaddlePaddle/PaddleOCR#13111] [Q&A]
转换指令为：python tools/export_model.py -c inference/ch_PP-OCR_v3_rec/config.yml -o Global.pretrained_model=./inference/ch_PP-OCR_v3_rec/best_accuracy  Global.save_inference_dir=./inferences_rec/

模型为官方“基于PP-OCRv3的液晶屏读数识别”模型：https://github.com/PaddlePaddle/PaddleOCR/blob/main/applications/%E6%B6%B2%E6%99%B6%E5%B1%8F%E8%AF%BB%E6%95%B0%E8%AF%86%E5%88%AB.md

报错内容为：    
shape=[None] + input_shape, dtype="float32"),
TypeError: can only concatenate list (not "NoneType") to list

环境为：
torch                     2.2.0+cu118
torchaudio                2.2.0+cu118
torchvision               0.17.0+cu118
paddleocr                 2.7.3
paddlepaddle-gpu          2.6.1
numpy                     1.23.5

只有识别模型无法转换，检测模型可以正常转换。

**Comment by wq881103** [comment]
v3版本的模型用paddleocr2.6.0版本的可以转换成功。
另外，使用paddlepaddle2.6.1转换出来的推理模型用于一些旧版软件会出错，比如umi-ocr。
想要兼容umi-ocr，需要将paddlepaddle降级，我试着降级了又与cuda版本不兼容，最后使用的paddlepaddle-gpu_2.4.2.post117，cuda11.7,cudnn8.8

**Reply by chunmengtu** [reply]
好的，我试一试，非常感谢Thanks♪(･ω･)ﾉ

**Reply by OeasyC** [reply]
v3版本模型训练自定义数据集2.7.0代码训练，2.7.0导出不了，2.8.0导出不了，2.6.0还是导出不了。
2.8.0训练，2.8.0可以导出，但是训练过程acc始终为0。

**Reply by LaiJinYan** [reply]
非常感谢，原来是版本问题
